vip-application-importer/src/main/java/fr/insalyon/creatis/vip/applicationimporter/server/business/VelocityUtils.java

```diff
@@ -85,4 +99,20 @@ public String createDocument( String tag, BoutiquesTool bt, Boolean isRunOnGrid,
 
         return stringWriter.toString();
     }
+
+    public String createDocument(String tag, BoutiquesTool bt, Boolean isRunOnGrid, String executionType, String vmTemplate) {
+        VelocityContext context = new VelocityContext();
+        context.put("tag", tag);
+        context.put("tool", bt);
+        context.put("isRunOnGrid", isRunOnGrid);
+        context.put("esc", new EscapeTool());
+        context.put("executionType", executionType);
+
+        StringWriter stringWriter = new StringWriter();
+
+        Template template = velocityEngine.getTemplate(vmTemplate);
+        template.merge(context, stringWriter);
+
+        return stringWriter.toString();
+    }
```

Comment:
The issue was already there before but it is not pretty to have the same method copy/pasted 4 times. 
That would be nice to improve that.